fdbserver/ClusterRecovery.actor.cpp

```diff
@@ -1080,7 +1079,15 @@ ACTOR Future<Void> readTransactionSystemState(Reference<ClusterRecoveryData> sel
 	// Sets self->configuration to the configuration (FF/conf/ keys) at self->lastEpochEnd
 
 	// Recover transaction state store
-	bool enableEncryptionForTxnStateStore = isEncryptionOpSupported(EncryptOperationType::TLOG_ENCRYPTION);
+	// If it's the first recovery the encrypt mode is not yet avilable so create the txn state store with encryption
+	// disabled. This is fine since we will not write any data to disk using this txn store.
+	state bool enableEncryptionForTxnStateStore = false;
+	if (self->controllerData && self->controllerData->encryptionAtRestMode.isValid() &&
```

Comment:
nit: do we need `isValid` check? `Promise` should be valid all the time, maybe just check for `isReady` should suffice?